tests/devices/test_default_qubit_torch.py

```diff
@@ -0,0 +1,1436 @@
+
+        with pytest.raises(
+            ValueError, match=r"BasisState parameter and wires must be of equal length"
+        ):
+            dev.apply([qml.BasisState(state, wires=[0, 1, 2])])
+
+    def test_invalid_basis_state(self, tol):
+        """Test that an exception is raised if the basis state is invalid"""
+        dev = DefaultQubitTorch(wires=4)
+        state = np.array([0, 0, 1, 2])
+
+        with pytest.raises(
+            ValueError, match=r"BasisState parameter must consist of 0 or 1 integers"
+        ):
+            dev.apply([qml.BasisState(state, wires=[0, 1, 2, 3])])
+
+    def test_qubit_state_vector(self, init_state, tol):
+        """Test qubit state vector application"""
+        dev = DefaultQubitTorch(wires=1)
+        state = init_state(1)
+
+        dev.apply([qml.QubitStateVector(state, wires=[0])])
+
+        res = dev.state
+        expected = state
+        assert isinstance(res, torch.Tensor)
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+    def test_full_subsystem_statevector(self, mocker):
+        """Test applying a state vector to the full subsystem"""
+        dev = DefaultQubitTorch(wires=["a", "b", "c"])
+        state = torch.tensor([1, 0, 0, 0, 1, 0, 1, 1], dtype=torch.complex128) / 2.0
+        state_wires = qml.wires.Wires(["a", "b", "c"])
+
+        spy = mocker.spy(dev, "_scatter")
+        dev._apply_state_vector(state=state, device_wires=state_wires)
+
+        assert np.all(np.reshape(dev._state.detach().numpy(), [-1]) == state.numpy())
+        spy.assert_not_called()
+
+    def test_partial_subsystem_statevector(self, mocker):
+        """Test applying a state vector to a subset of wires of the full subsystem"""
+        dev = DefaultQubitTorch(wires=["a", "b", "c"])
+        state = torch.tensor([1, 0, 1, 0], dtype=torch.complex128) / np.sqrt(2.0)
+        state_wires = qml.wires.Wires(["a", "c"])
+
+        spy = mocker.spy(dev, "_scatter")
+        dev._apply_state_vector(state=state, device_wires=state_wires)
+        res = torch.reshape(torch.sum(dev._state, axis=(1,)), [-1])
+
+        assert np.all(res.detach().numpy() == state.detach().numpy())
+        spy.assert_called()
+
+    def test_invalid_qubit_state_vector_size(self):
+        """Test that an exception is raised if the state
+        vector is the wrong size"""
+        dev = DefaultQubitTorch(wires=2)
+        state = np.array([0, 1])
+
+        with pytest.raises(ValueError, match=r"State vector must be of length 2\*\*wires"):
+            dev.apply([qml.QubitStateVector(state, wires=[0, 1])])
+
+    def test_invalid_qubit_state_vector_norm(self):
+        """Test that an exception is raised if the state
+        vector is not normalized"""
+        dev = DefaultQubitTorch(wires=2)
+        state = np.array([0, 12])
+
+        with pytest.raises(ValueError, match=r"Sum of amplitudes-squared does not equal one"):
+            dev.apply([qml.QubitStateVector(state, wires=[0])])
```

Comment:
This should help get coverage on the case where we expect a Torch Tensor to be passed.
```suggestion
    @pytest.mark.parametrize("state", [np.array([0, 12]), torch.tensor([1., -1.], requires_grad=True)])
    def test_invalid_qubit_state_vector_norm(self, state):
        """Test that an exception is raised if the state
        vector is not normalized"""
        dev = DefaultQubitTorch(wires=2)

        with pytest.raises(ValueError, match=r"Sum of amplitudes-squared does not equal one"):
            dev.apply([qml.QubitStateVector(state, wires=[0])])
```